src/main/java/com/checkmarx/intellij/tool/window/CxToolWindowPanel.java

```diff
@@ -131,6 +134,19 @@ private void drawMainPanel() {
                                                                                       .findFirst()
                                                                                       .orElse(null);
 
+        // update scan buttons visibility based on tenant settings
+        CompletableFuture.supplyAsync(() -> {
+            try {
+                return TenantSetting.isScanAllowed();
+            } catch (Exception e) {
+                return null;
```

Comment:
Should we log and the exception and return false?

Then TenantSetting.isScanAllowed() could return a primitive boolean instead of object as well and remove the null check on the consumer.

Reply:
Done